### DISCUSSION TITLE
mwc.pl seems to not generate all files under windows for a java target

### DISCUSSION THREAD

**Discussion by bertrandbado** [OpenDDS/OpenDDS#4819] [Q&A]
Hi,

I am trying to generate Java libraries on Windows.

I followed the following steps:
* configure --java
* build DDS_TAOv2.sln (lots of warnings, some errors but final status was ok)
* Then I wrote a very simple idl file and copied a sample mpc file
* I generate Foo_Export.h using %ACE_ROOT%\bin\generate_export_file.pl Foo > Foo_Export.h
* Then I generate a Visual Studio solution using: %ACE_ROOT%\bin\mwc.pl -type vs2022

When I try to build this solution, there are a lot of missing Java files because the script only generated the .sln, .vcxproj and .vcxproj.filters .

Did I miss a step to get a complete solution?

Thank you for your help.

**Comment by mitza-oci** [comment]
Which files were generated by this last build step?  If you try the build command again, do more files get generated?
There is an issue with Visual Studio where not all the dependencies are built.

**Reply by bertrandbado** [reply]
Hi Mitza-oci,

Thank you for your answer. 
This is files generated by the last command : 
![image](https://github.com/user-attachments/assets/1facab89-2609-4af3-8dec-7747d9e361c2)

I tried to  rebuild the entire DDS_TAOv2.sln, I got 0 error 75381 warning. 
What do you mean by trying the build command again ?

**Reply by mitza-oci** [reply]
Those files (3 of the 4 anyway) are generated by mwc.pl.  What is generated when you run the build, as in "When I try to build this solution..."?

**Reply by bertrandbado** [reply]
Hi, 

Thank you, I don't know what I miss before but when I follow the steps describe here, files are generated at the build time. 

Unfortunatly (but it's not exactly the same issue) FOO_Export.h contains errors. 
The first one is  C2008 : '.' unexpected in a macro definition

This is the concerned line : #define TOTO_STATION.IDL_EXPORT_H

**Reply by mitza-oci** [reply]
The command line argument for `generate_export_file.pl` shouldn't contain a `.` since it's used as an identifier.
